lib/beautiful/xresources.lua

```diff
@@ -83,8 +83,16 @@ function xresources.get_dpi(s)
         if awesome and awesome.xrdb_get_value then
             xresources.dpi = tonumber(awesome.xrdb_get_value("", "Xft.dpi"))
         end
+        -- Following Keith Packard's whitepaper on Xft,
+        -- https://keithp.com/~keithp/talks/xtc2001/paper/xft.html#sec-editing
+        -- the proper fallback for Xft.dpi is the vertical DPI reported by
+        -- the X server. This will generally be 96 on Xorg, unless the user
+        -- has configured it differently
         if not xresources.dpi then
-            xresources.dpi = 96
+            local root = require("root")
+            _, h = root.size()
+            _, hmm = root.size_mm()
+            xresources.dpi = round(h*25.4/hmm)
```

Comment:
Oh, actually it fails on Travis while running unit tests, which is half-unrelated. Wow, I did not know that the C code makes things like `root` `require`able....